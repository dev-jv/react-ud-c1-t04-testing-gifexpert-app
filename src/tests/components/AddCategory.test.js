import React from 'react';
import { AddCategory } from '../../components/AddCategory';
import { shallow } from 'enzyme';

// eslint-disable-next-line no-unused-vars
import colors from 'colors';

// eslint-disable-next-line jest/valid-title
describe('Testing in <AddCategory />'.bgBlue.white, () => {

    const setCategories = () => {};
    const wrapper = shallow(<AddCategory setCategories = {setCategories} />);

    test('1: Should show correctly the component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('1: Should change the text box', () => {
        const input = wrapper.find('input');
        const value = 'new value';
        input.simulate('change', {target: { value }});
        expect(wrapper.find('p').text().trim()).toBe(value);
    });

});
