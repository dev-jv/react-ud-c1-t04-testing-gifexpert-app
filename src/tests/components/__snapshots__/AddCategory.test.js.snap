// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[37m[44mTesting in <AddCategory />[49m[39m 1: Should show correctly the component 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
     
     
  </p>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
